tests/Integration/UI/SettingsPageTest.php

```diff
@@ -0,0 +1,91 @@
+<?php
+/**
+ * Settings page tests.
+ *
+ * @package Parsely
+ */
+
+declare(strict_types=1);
+
+namespace Parsely\Tests\Integration\UI;
+
+use Parsely\Parsely;
+use Parsely\Tests\Integration\TestCase;
+use Parsely\UI\Settings_Page;
+
+/**
+ * Row actions tests.
+ *
+ * @since 3.1.0
+ */
+final class SettingsPageTest extends TestCase {
+	/**
+	 * Internal variable.
+	 *
+	 * @var Settings_Page $settings_page Holds the Settings_Page object.
+	 */
+	private static $settings_page;
+	/**
+	 * Internal variable.
+	 *
+	 * @var Parsely $parsely Holds the Parsely object.
+	 */
+	private static $parsely;
+
+	/**
+	 * The setup run before each test.
+	 */
+	public function set_up(): void {
+		parent::set_up();
+		self::$parsely       = new Parsely();
+		self::$settings_page = new Settings_Page( self::$parsely );
+	}
+
+	/**
+	 * Check that default tracking values get saved.
+	 *
+	 * @since 3.1.0
+	 *
+	 * @covers \Parsely\UI\Settings_Page::validate_options
+	 * @group ui
+	 */
+	public function test_validate_unique_tracking_values_succeeds(): void {
+		// Initializations.
+		$expected = self::$parsely->get_options();
+		$options  = self::$parsely->get_options();
+
+		// Default tracking values.
+		$options['track_post_types'] = array( 'post' );
+		$options['track_page_types'] = array( 'page' );
+		self::assertSame( self::$settings_page->validate_options( $options ), $expected );
+	}
+
+	/**
+	 * Check that validate_options() method will not allow duplicate tracking.
+	 *
+	 * @since 3.1.0
+	 *
+	 * @covers \Parsely\UI\Settings_Page::validate_options
+	 * @group ui
+	 */
+	public function test_validate_duplicate_tracking_values_returns_errors(): void {
+		// Initializations.
+		$expected = self::$parsely->get_options();
+		$options  = self::$parsely->get_options();
+
+		// Duplicate selection in Post Types.
+		$options['track_post_types'] = array( 'post', 'page' );
+		$options['track_page_types'] = array( 'page' );
+		self::assertSame( self::$settings_page->validate_options( $options ), $expected );
+
+		// Duplicate selection in Page Types.
+		$options['track_post_types'] = array( 'post' );
+		$options['track_page_types'] = array( 'post', 'page' );
+		self::assertSame( self::$settings_page->validate_options( $options ), $expected );
```

Comment:
I would separate the three assertions on three separate tests. In the event of a test failing, we would get better granularity.